src/components/absorber/module.cairo

```diff
@@ -0,0 +1,402 @@
+    }
+
+    #[test]
+    #[available_gas(20_000_000)]
+    fn test_final_time_zero_not_set() {
+        // [Setup]
+        let mut state = STATE();
+        // [Assert] Final time = times[-1]
+        let time = Absorber::AbsorberImpl::get_final_time(@state, 0);
+        assert(time == 0, 'Wrong time');
+    }
+
+    #[test]
+    #[available_gas(20_000_000)]
+    fn test_final_absorption() {
+        // [Setup]
+        let mut state = STATE();
+        let times: Span<u64> = array![1651363200, 1659312000, 1667260800, 1675209600, 1682899200]
+            .span();
+        let absorptions: Span<u64> = array![0, 1179750, 2359500, 3539250, 4719000].span();
+        let ton_equivalent = 1000000;
+        Absorber::AbsorberImpl::set_absorptions(ref state, 0, times, absorptions, ton_equivalent);
+        // [Assert] Final absorption = absorptions[-1]
+        let absorption = Absorber::AbsorberImpl::get_final_absorption(@state, 0);
+        assert(absorption == *absorptions.at(absorptions.len() - 1), 'Wrong absorption');
+    }
+
+    #[test]
+    #[available_gas(20_000_000)]
+    fn test_final_absorption_zero_not_set() {
+        // [Setup]
+        let mut state = STATE();
+        // [Assert] Final absorption = absorptions[-1]
+        let absorption = Absorber::AbsorberImpl::get_final_absorption(@state, 0);
+        assert(absorption == 0, 'Wrong absorption');
+    }
+
+    #[test]
+    #[available_gas(20_000_000)]
+    #[should_panic(expected: ('Times and absorptions mismatch',))]
+    fn test_set_absorptions_revert_mismatch() {
+        // [Setup]
+        let mut state = STATE();
+        let times: Span<u64> = array![1651363200, 1659312000, 1667260800, 1675209600, 1682899200]
+            .span();
+        let absorptions: Span<u64> = array![].span();
+        let ton_equivalent = 0;
+        Absorber::AbsorberImpl::set_absorptions(ref state, 0, times, absorptions, ton_equivalent);
+    }
+
+    #[test]
+    #[available_gas(20_000_000)]
+    #[should_panic(expected: ('Inputs cannot be empty',))]
+    fn test_set_absorptions_revert_empty() {
+        // [Setup]
+        let mut state = STATE();
+        let times: Span<u64> = array![].span();
+        let absorptions: Span<u64> = array![].span();
+        let ton_equivalent = 0;
+        Absorber::AbsorberImpl::set_absorptions(ref state, 0, times, absorptions, ton_equivalent);
+    }
+
+    #[test]
+    #[available_gas(20_000_000)]
+    #[should_panic(expected: ('Ton equivalent must be positive',))]
+    fn test_set_absorptions_revert_not_positive() {
+        // [Setup]
+        let mut state = STATE();
+        let times: Span<u64> = array![1651363200, 1659312000, 1667260800, 1675209600, 1682899200]
+            .span();
+        let absorptions: Span<u64> = array![0, 1179750, 2359500, 3539250, 4719000].span();
+        let ton_equivalent = 0;
+        Absorber::AbsorberImpl::set_absorptions(ref state, 0, times, absorptions, ton_equivalent);
+    }
+}
```

Comment:
is calling set_absorption_twice possible?

Reply:
yes, otherwize I would set it in the initializer